var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -873,6 +883,43 @@ class Opencv(CMakePackage, CudaPackage):
     conflicts("+win32ui", when="platform=linux", msg="Windows only")
     conflicts("+win32ui", when="platform=cray", msg="Windows only")
 
+    @classmethod
+    def determine_version(cls, lib):
+        ver = None
+        for ext in library_extensions:
+            pattern = None
+            if ext == 'dylib':
+                # Darwin switches the order of the version compared to Linux
+                pattern = re.compile(r'lib(\S*?)_(\S*)\.(\d+\.\d+\.\d+)\.%s' %
+                                     ext)
+            else:
+                pattern = re.compile(r'lib(\S*?)_(\S*)\.%s\.(\d+\.\d+\.\d+)' %
+                                     ext)
+            match = pattern.search(lib)
+            if match:
+                ver = match.group(3)
+        return ver
+
+    @classmethod
+    def determine_variants(cls, libs, version_str):
+        results = []
+        variants = []
+        for lib in libs:
+            for ext in library_extensions:
+                pattern = None
+                if ext == 'dylib':
+                    # Darwin switches the order of the version compared to Linux
+                    pattern = re.compile(r'lib(\S*?)_(\S*)\.(\d+\.\d+\.\d+)\.%s' %
+                                         ext)
+                else:
+                    pattern = re.compile(r'lib(\S*?)_(\S*)\.%s\.(\d+\.\d+\.\d+)' %
+                                         ext)
+                match = pattern.search(lib)
+                if match and not match.group(2) == 'core':
+                    variants.append('+' + match.group(2))
+        results.append(' '.join(variants))
+        return results
```

Comment:
Shouldn't this be:
```suggestion
        return ' '.join(variants)
```
?